contrib/format-ltsv/src/main/java/org/apache/drill/exec/store/ltsv/LTSVRecordReader.java

```diff
@@ -0,0 +1,165 @@
+  private DrillBuf buffer;
+
+  private VectorContainerWriter writer;
+
+  public LTSVRecordReader(FragmentContext fragmentContext, Path path, DrillFileSystem fileSystem,
+                          List<SchemaPath> columns) throws OutOfMemoryException {
+    this.inputPath = path.toString();
+    try {
+      this.fsStream = fileSystem.open(path);
+      this.reader = new BufferedReader(new InputStreamReader(fsStream.getWrappedStream(), StandardCharsets.UTF_8));
+      this.buffer = fragmentContext.getManagedBuffer();
+      setColumns(columns);
+
+    } catch (IOException e) {
+      String msg = "Failed to open open input file: " + inputPath;
+      throw UserException.dataReadError(e).message(msg).build(logger);
+    }
+  }
+
+  @Override
+  protected Collection<SchemaPath> transformColumns(Collection<SchemaPath> projected) {
+    Set<SchemaPath> transformed = Sets.newLinkedHashSet();
+    if (!isStarQuery()) {
+      for (SchemaPath column : projected) {
+        transformed.add(column);
+      }
+    } else {
+      transformed.add(SchemaPath.STAR_COLUMN);
+    }
+    return transformed;
+  }
+
+  public void setup(final OperatorContext context, final OutputMutator output) throws ExecutionSetupException {
+    this.writer = new VectorContainerWriter(output);
+  }
+
+  public int next() {
+    this.writer.allocate();
+    this.writer.reset();
+
+    int recordCount = 0;
+
+    try {
+      BaseWriter.MapWriter map = this.writer.rootAsMap();
+      String line = null;
+
+      while (recordCount < MAX_RECORDS_PER_BATCH && (line = this.reader.readLine()) != null) {
+        // Skip empty lines
+        if (line.trim().length() == 0) {
+          continue;
+        }
+
+        List<String[]> fields = Lists.newArrayList();
+        for (String field : line.split("\t")) {
+          int index = field.indexOf(":");
+          if (index <= 0) {
+            throw new ParseException("Invalid LTSV format: " + inputPath + "\n" + (recordCount + 1) + ":" + line, 0);
+          }
+
+          String fieldName = field.substring(0, index);
+          String fieldValue = field.substring(index + 1);
+          if (selectedColumn(fieldName)) {
+            fields.add(new String[]{fieldName, fieldValue});
+          }
+        }
+
+        if (fields.size() == 0) {
+          continue;
+        }
+
+        this.writer.setPosition(recordCount);
+        map.start();
+
+        for (String[] field : fields) {
+          byte[] bytes = field[1].getBytes(StandardCharsets.UTF_8);
+          this.buffer = this.buffer.reallocIfNeeded(bytes.length);
+          this.buffer.setBytes(0, bytes, 0, bytes.length);
+          map.varChar(field[0]).writeVarChar(0, bytes.length, buffer);
+        }
+
+        map.end();
+        recordCount++;
+      }
+
+      this.writer.setValueCount(recordCount);
+      return recordCount;
+
+    } catch (final Exception e) {
+      String msg = "Failure while reading messages from LTSV. Recordreader was at record: " + (recordCount + 1);
```

Comment:
```suggestion
      String msg = "Failure while reading messages from LTSV. RecordReader was at record: " + (recordCount + 1);
```

Also you can add path to the exception.